.github/workflows/html5lib-tests.yml

```diff
@@ -0,0 +1,195 @@
+name: HTML API html5lib Tests
+
+on:
+  push:
+    branches:
+      - trunk
+      # The html5lib tests were introduced in WordPress 6.6.
+      - '6.[6-9]'
+      - '[7-9].[0-9]'
+      - '[1-9][0-9]+.[0-9]'
+    tags:
+      - '6.[6-9]'
+      - '6.[6-9].[0-9]+'
+      - '[7-9].[0-9]'
+      - '[7-9].[0-9].[0-9]+'
+      - '[1-9][0-9]+.[0-9]'
+      - '[1-9][0-9]+.[0-9].[0-9]+'
+  pull_request:
+    branches:
+      - trunk
+      - '6.[6-9]'
+      - '[7-9].[0-9]'
+      - '[1-9][0-9]+.[0-9]'
+    paths:
+      # Any change to a PHP file in the HTML API should run checks.
+      - 'src/wp-includes/html-api/**.php'
+      - 'tests/phpunit/tests/html-api/wpHtmlProcessorHtml5lib.php'
+      - 'tests/phpunit/data/html5lib-tests/**.dat'
+  workflow_dispatch:
+
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+  # The concurrency group contains the workflow name and the branch name for pull requests
+  # or the commit hash for any other events.
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+# Disable permissions for all available scopes by default.
+# Any needed permissions should be configured at the job level.
+permissions: {}
+
+jobs:
+  # Runs the HTML API html5lib PHPUnit tests for WordPress.
+  #
+  # Performs the following steps:
+  # - Sets environment variables.
+  # - Checks out the repository.
+  # - Sets up Node.js.
+  # - Sets up PHP.
+  # - Installs Composer dependencies.
+  # - Installs npm dependencies
+  # - Logs general debug information about the runner.
+  # - Logs Docker debug information (about the Docker installation within the runner).
+  # - Starts the WordPress Docker container.
+  # - Logs the running Docker containers.
+  # - Logs debug information about what's installed within the WordPress Docker containers.
+  # - Install WordPress within the Docker container.
+  # - Run the HTML API html5lib PHPUnit tests.
+  # - Ensures version-controlled files are not modified or deleted.
+  run-html-api-html5lib-tests:
+    name: PHP ${{ matrix.php }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    if: ${{ github.repository == 'WordPress/wordpress-develop' || github.event_name == 'pull_request' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        php: [ '7.2', '7.3', '7.4', '8.0', '8.1', '8.2', '8.3' ]
```

Comment:
I  don't think either would be valuable here since these tests are extremely close to actual unit tests. None of the HTML API code interacts with the database nor does it rely on any code which is different for multisite.